internal/api/api.go

```diff
@@ -402,6 +417,22 @@ func (a *API) handleError(writer http.ResponseWriter, errToHandle error, fallbac
 	return
 }
 
+func (a *API) ServiceBindingsSecrets(sbs *types.ServiceBindings) responses.ServiceBindingSecret {
+	serviceBindingsSecrets := make(responses.ServiceBindingSecret, 0)
+	for _, sb := range sbs.Items {
+		secrets, err := a.secretsForGivenServiceBindingID(sb.ID)
+		if err != nil {
+			a.logger.Error("failed to get secrets for service binding", "service binding id", sb.ID, "error", err)
+			continue
+		}
+		if len(secrets.Items) > 0 {
+			serviceBindingsSecrets[sb.ID] = &secrets.Items[0]
```

Comment:
The same questions. It the length either 0 or 1?

Reply:
See my answer for L280: https://github.com/kyma-project/btp-manager/pull/802#discussion_r1713336200
> A Service Binding should have only one secret with credentials, but the method to fetch the secret uses labels to get labeled secrets and there may be many copies of the secret in various namespaces. We just need one of them.